index.js

```diff
@@ -71,16 +71,16 @@ function flameGraph (opts) {
   var hoverFrame = null
   var currentAnimation = null
 
-  // Use custom coloring function if defined
+  // Overridable functions. Use custom function if passed in, default if undefined, or, do nothing (or neutral fallback) if passed null
   var colorHash = (opts.colorHash === undefined) ? defaultColorHash : (d, decimalAdjust, allSamples, tiers) => opts.colorHash ? opts.colorHash(stackTop, { d, decimalAdjust, allSamples, tiers }) : frameColors.fill
 
-  // Use custom text label rendering function if defined
   var renderLabel = (opts.renderLabel === undefined) ? defaultRenderLabel : (context, node, x, y, width) => opts.renderLabel && opts.renderLabel(c, { context, node, x, y, width })
 
-  // Use custom tooltip rendering function if defined
   var renderTooltip = (opts.renderTooltip === undefined) ? defaultRenderTooltip : node => opts.renderTooltip && opts.renderTooltip(node)
 
-  // Use custom handler for clicks on canvas if defined; preserves default `this` as being the DOM object
+  var stackBoxGlobals = { frameHeight: c, STATE_HOVER, STATE_UNHOVER, STATE_IDLE, frameColors, colorHash }
```

Comment:
`c` can be changed at runtime using `chart.cellHeight(newC)`, so this object should maybe be regenerated when it changes / built whenever custom `renderStackFrameBox` is called

it would also be nice to rename the `c` variable to something like frameHeight internally but that's also a separate thing

We could get rid of the `STATE_*` variables by using strings instead of numeric constants, then do `state === 'hover'` instead of `STATE_HOVER`. alternatively the constants could be exposed on `module.exports`. Then the first parameter to renderStackFrameBox just contains d3-fg options and the second parameter contains node info?

Reply:
Good spot. I think a better way is to hoover it up alongside x, y and width as `height` in a separate `rect` argument, which matches other `rect` objects. See latest commit.

Would love to replace all the one-character variable names like `c` but there's a big risk of regressions and bugs (few tests in repo, can't even `ctrl-f` as a sanity check...) so it's something we should do when we've got more time.

Likewise refactoring the STATE_s isn't a big priority right now.